zipkin-collector/core/pom.xml

```diff
@@ -34,7 +34,7 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
-      <version>${slf4j.version}</version>
+      <version>2.0.11</version>
```

Comment:
Looking for better option here: the `slf4j-test` (below) does not support slf4j 1.7

Reply:
maybe revert the test changes here to before the switch to 2.0?

Reply:
wait that probably brings back the junit 4 dep ;) maybe write a custom in-memory appender?

Reply:
or.. if there's a way to override slf4j version, but only in tests. not sure if that is possible in maven or not

Reply:
other ways:
* make an invoker module and move this test there, in that module use slf4j 2 and this extension, as then it doens't taint the dep tree.
* let this be 2.x but have zipkin-server exclude its transitive dep on slf4j in its pom

about the latter, I wonder why it works.. I guess zipkin-server is already setting the version? In any case we should add a comment if we end up leaving this as 2

Reply:
a last option would be to move this to an invoker test in zipkin-server, then in that test we can also check armeria ES logging works ;)

Reply:
@codefromthecrypt I reverted it back to `uk.org.lidalia:slf4j-test` (I know it is not supported by that's the only one that still works with 1.7.x)

Reply:
> about the latter, I wonder why it works.. I guess zipkin-server is already setting the version? In any case we should add a comment if we end up leaving this as 2

We also kind of mix the dependencies there, for example for Spring Boot, we replace Log4j2 -> Slf4j2 with Log4j2 -> Slf4j:

```
<!-- Use log4j 2 as default logging implementation -->
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-log4j2</artifactId>
      <version>${spring-boot.version}</version>
      <exclusions>
        <!-- spring boot configures slf4j2 not slf4j -->
        <exclusion>
          <groupId>org.apache.logging.log4j</groupId>
          <artifactId>log4j-slf4j2-impl</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
    <!-- spring boot configures slf4j2 not slf4j -->
    <dependency>
      <groupId>org.apache.logging.log4j</groupId>
      <artifactId>log4j-slf4j-impl</artifactId>
      <version>${log4j.version}</version>
    </dependency>
```

Reply:
in reflection reverting to the old slf4j-test is fine because it isn't the only thing still depending on old junit, testcontainers does, too

Reply:
Exactly, we could also do that:
 - revert to Slf4j 1.7.x to get things in order (with this pull request)
 - look on the ways to bring Slf4j 2.x, may need some upstream pull requests